src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -256,6 +347,9 @@ def _validate_vm_create_nics(namespace):
 
 def _validate_vm_create_auth(namespace):
 
+    if namespace.storage_profile == StorageProfile.ManagedSpecializedOSDisk:
```

Comment:
We should check that auth parameters are not supplied since they would be ignored.